tap_stripe/__init__.py

```diff
@@ -1033,25 +1045,36 @@ def sync_event_updates(stream_name, is_sub_stream):
         # The events stream returns results in descending order, so we
         # cannot bookmark until the entire page is processed
         date_window_start = date_window_end
-        date_window_end = date_window_end + date_window_size
+        date_window_end = date_window_end + events_date_window_size
 
-        # Write the parent bookmark value only when the parent is selected
-        if not is_sub_stream:
-            singer.write_bookmark(Context.state,
-                                stream_name + '_events',
-                                'updates_created',
-                                max_created)
-            singer.write_state(Context.state)
+        # Write bookmark for parent or child stream if it is selected
+        write_bookmark_for_event_updates(is_sub_stream, stream_name, sub_stream_name, max_created)
+
+    max_created = max(max_created, sync_start_time - events_date_window_size)
```

Comment:
Add comment for this logic

Reply:
Added comment to explain the bookmark logic.